client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt

```diff
@@ -297,63 +338,486 @@ public class MarketPageModel(client: DesktopClient) {
+ * @param share the share to represent
+ * @param previousPrice the previous price of this share
+ */
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+private fun ShareItem(
+    model: MarketPageModel,
+    share: Share,
+    previousPrice: Money?
+) {
+    Box(
+        modifier = Modifier
+            .clip(MaterialTheme.shapes.small)
+            .bottomBorder()
+    ) {
+        ListItem(
+            modifier = Modifier
+                .height(60.dp)
+                .clickable(
+                    enabled = true,
+                    onClick = {
+                        model.selectedShare(share.id)
+                    }
+                ),
+            headlineText = {
+                ShareItemContent(share, previousPrice)
+            },
+        )
+    }
+}
+
+/**
+ * Represents the main `ListItem` content with data about the share.
+ *
+ * @param share the share to show inside the item
+ * @param previousPrice the previous price of this share
+ */
+@Composable
+private fun ShareItemContent(
+    share: Share,
+    previousPrice: Money?
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxSize()
+    ) {
+        Column(
+            modifier = Modifier
+                .weight(2F)
+                .fillMaxHeight(),
+            horizontalAlignment = Alignment.Start,
+            verticalArrangement = Arrangement.Center
+        ) {
+            Text(share.companyName, style = MaterialTheme.typography.bodySmall)
+        }
+        Column(
+            modifier = Modifier
+                .weight(1F)
+                .fillMaxHeight(),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = Arrangement.Center
+        ) {
+            Text(share.price.asReadableString(), style = MaterialTheme.typography.headlineSmall)
+            val priceDifference = definePriceDifferenceConfig(share.price, previousPrice)
+            PriceDifferenceCard(priceDifference)
+        }
+    }
+}
+
+/**
+ * Represents the main information about the share and ways to interact with it.
+ *
+ * @param share the share to represent
+ * @param previousPrice the previous price of this share
+ * @param model the model of the market page
+ */
+@Composable
+private fun ShareProfile(
+    share: Share,
+    previousPrice: Money?,
+    model: MarketPageModel
+) {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(15.dp),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        ShareLogo(share)
+        Text(
+            text = share.companyName,
+            style = MaterialTheme.typography.labelMedium,
+            modifier = Modifier
+                .padding(top = 10.dp)
+        )
+        SharePrice(share.price, previousPrice)
+        ButtonSection(
+            model = model,
+            share = share
+        )
+    }
+}
+
+/**
+ * Show information about the share price.
```

Comment:
In terms of grammar, it should be "Shows". But to men, "Displays"  would sound better.